### PR TITLE
Add ping_quorum and get_peer_info support calls

### PR DESCRIPTION
**Pull request by jtuple**
`riak_ensemble_peer:ping_quorum/1` checks an ensemble for an active
quorum, returning the current ensemble leader as well as the list of
peers supporting the current quorum.

`riak_ensemble_manager:get_peer_info/2` returns information about a
given ensemble peer: the peer's current FSM state, if it is or is not
trusted, and it's current epoch.

Sibling pull-request: basho/riak_kv#979

### CODE REVIEW

**Comment by engelsanchez** [comment]
I'm taking this review

File: src/riak_ensemble_peer.erl [hunk 1]
```diff
@@ -641,25 +662,26 @@ leading(check_quorum, From, State) ->
...
-                                       %% io:format("met: ~p~n", [Replies]),
-                                       length(Replies) + Extra;
-                                   {timeout, _Replies} ->
-                                       %% io:format("timeout~n"),
-                                       Extra
-                               end,
-                       gen_fsm:reply(From, Count)
+                       Result = case wait_for_quorum(Future) of
+                                    {quorum_met, Replies} ->
+                                        %% io:format("met: ~p~n", [Replies]),
```
Comment (reiddraper) [inline src/riak_ensemble_peer.erl:678]: this snuck in

File: src/riak_ensemble_peer.erl [hunk 2]
```diff
@@ -641,25 +662,26 @@ leading(check_quorum, From, State) ->
...
+                                        Extra ++ Replies;
+                                    {timeout, _Replies} ->
+                                        %% io:format("timeout~n"),
```
Comment (reiddraper) [inline src/riak_ensemble_peer.erl:681]: this snuck in

File: src/riak_ensemble_peer.erl (hunk 1 above)
Comment (jtuple) [inline src/riak_ensemble_peer.erl:678]: It didn't really sneak in. That line was in the original code as shown in the diff. Didn't want to clutter PR with changes unrelated to the actual change made.

**Comment by engelsanchez** [comment]
+1 f547fb5

**Comment by jtuple** [comment]
@borshop merge

File: src/riak_ensemble_peer.erl (hunk 1 above)
Comment (reiddraper) [inline src/riak_ensemble_peer.erl:678]: ah, cool. wfm then

**Pull request merged**